### PR TITLE
Search Bar

### PR DESCRIPTION
**Pull request by Meghanaskanagal**
1) Search bar changes for Homepage.

### CODE REVIEW

**Comment by invincibleJai** [comment]
@Meghanaskanagal Gif showing the changes or screenshot would be good to have and have one single commit   would be good  so better to  squash

File: run_local.sh [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/sh
 
 export REACT_APP_BACKEND_URL="http://localhost:8080"
-export REACT_APP_MAPBOX_TOKEN="your_token_here"
+export REACT_APP_MAPBOX_TOKEN="pk.eyJ1IjoibWVnaGFuYXNrYW5hZ2FsIiwiYSI6ImNrMXExdmU5ZjEyYWczY3FvZmw2dm9oamwifQ.qDuxo-hNK1XTwOGDRsvSTA"
```
Comment (invincibleJai) [inline run_local.sh]: @Meghanaskanagal do we really want it?

File: src/App.css [hunk 2]
```diff
@@ -15,3 +15,14 @@
...
 }
+
+.searchbar{
+  max-width: 35em;
+  margin-right: 1em;
+  align-items: center;
+  border-radius: 35px;
+  height: 3em !important;
+  outline: 0;
+  font-size: 20px;
```
Comment (invincibleJai) [inline src/App.css]: pf variables would be good to have

File: src/Display.tsx [hunk 3]
```diff
@@ -115,6 +114,8 @@ interface VictimDetailProps {
 
 const VictimDetail: React.FC<VictimDetailProps> = props => {
   const [address, setAddress] = useState("");
+  const [neighbourAddress, setNeighbourAddress] = useState("");
+  console.table(props.data);
```
Comment (invincibleJai) [inline src/Display.tsx]: is this needed?

File: src/Display.tsx [hunk 4]
```diff
@@ -124,8 +125,18 @@ const VictimDetail: React.FC<VictimDetailProps> = props => {
   )
     .then(response => response.json())
     .then(jsonData => {
+      debugger
```
Comment (invincibleJai) [inline src/Display.tsx]: is this needed?

File: src/Display.tsx [hunk 5]
```diff
@@ -124,8 +125,18 @@ const VictimDetail: React.FC<VictimDetailProps> = props => {
...
       if (jsonData.features.length) {
+        console.table(jsonData.features);
```
Comment (invincibleJai) [inline src/Display.tsx]: is this needed?

File: src/Display.tsx [hunk 6]
```diff
@@ -124,8 +125,18 @@ const VictimDetail: React.FC<VictimDetailProps> = props => {
...
+        let neighbouringLocations = [];
+        for (let i = 1; i < 3; i++) {
```
Comment (invincibleJai) [inline src/Display.tsx]: why 3?

File: src/Display.tsx [hunk 7]
```diff
@@ -124,8 +125,18 @@ const VictimDetail: React.FC<VictimDetailProps> = props => {
...
+          neighbouringLocations.push(jsonData.features[i].place_name + "\n\n");
```
Comment (invincibleJai) [inline src/Display.tsx]: can't we loop through `jsonData.features` , what are we trying to achieve

File: src/Display.tsx [hunk 8]
```diff
@@ -124,8 +125,18 @@ const VictimDetail: React.FC<VictimDetailProps> = props => {
...
+        }
+        console.log(neighbouringLocations);
```
Comment (invincibleJai) [inline src/Display.tsx]: do we need this

File: src/Display.tsx [hunk 9]
```diff
@@ -152,6 +163,13 @@ const VictimDetail: React.FC<VictimDetailProps> = props => {
                 <FlexItem>Phone:</FlexItem>
                 <FlexItem>Needs First Aid:</FlexItem>
                 <FlexItem>Location:</FlexItem>
+                {props.data.status === "REPORTED" && (
+                  <FlexItem>Neighbour Location's:</FlexItem>
+                )}
+                {props.data.status === "ASSIGNED" && (
+                  <FlexItem>Neighbour Location's:</FlexItem>
+                )}
+
```
Comment (invincibleJai) [inline src/Display.tsx]: can we have REPORTED, ASSIGNED as `const`

File: src/Display.tsx [hunk 10]
```diff
@@ -165,8 +183,19 @@ const VictimDetail: React.FC<VictimDetailProps> = props => {
                 </FlexItem>
                 <FlexItem>{props.data.numberOfPeople}</FlexItem>
                 <FlexItem>{props.data.victimPhoneNumber}</FlexItem>
-                <FlexItem>{String(props.data.medicalNeeded)}</FlexItem>
+                {props.data.medicalNeeded === true && (
```
Comment (invincibleJai) [inline src/Display.tsx]: can data be undefined?

File: src/Display.tsx [hunk 11]
```diff
@@ -189,16 +218,24 @@ interface DisplayListProps {
 }
 
 const DisplayList: React.FC<DisplayListProps> = props => {
-  let content = props.dataArray.map((val: any, key: number) => (
+  let content:any;
+  console.table(props.dataArray);
```
Comment (invincibleJai) [inline src/Display.tsx]: do we need it?

File: src/Display.tsx [hunk 12]
```diff
@@ -189,16 +218,24 @@ interface DisplayListProps {
...
+  if (props.dataArray.length === 0) {
```
Comment (invincibleJai) [inline src/Display.tsx]: can `dataArray` be undefined?

File: src/Display.tsx [hunk 13]
```diff
@@ -189,16 +218,24 @@ interface DisplayListProps {
...
+    if(props.isReady === false){
```
Comment (invincibleJai) [inline src/Display.tsx]: do we really need `===`

File: src/Display.tsx [hunk 14]
```diff
@@ -189,16 +218,24 @@ interface DisplayListProps {
...
+      content = <Alert
+      variant="danger"
+      isInline
+      title="No data available"
+    />;
+    }else{
+      content = <p></p>;
```
Comment (invincibleJai) [inline src/Display.tsx]: why `<p></p>`

File: src/Display.tsx [hunk 15]
```diff
@@ -189,16 +218,24 @@ interface DisplayListProps {
...
+    }
+  }else{
+    content = props.dataArray.map((val: any, key: number) => (
```
Comment (invincibleJai) [inline src/Display.tsx]: can we have `type` for val

File: src/Display.tsx [hunk 16]
```diff
@@ -216,4 +253,4 @@ const DisplayList: React.FC<DisplayListProps> = props => {
   );
 };
 
-export default DisplayList;
+export default DisplayList;
```
Comment (invincibleJai) [inline src/Display.tsx]: new line pls

File: src/Search.tsx [hunk 17]
```diff
@@ -19,19 +20,16 @@ const SearchForm: React.FC<SearchFormProps> = props => {
 
   return (
     <PageSection>
-      <form onSubmit={handleSubmit}>
+      <form style={{ textAlign: "center" }} onSubmit={handleSubmit}>
```
Comment (invincibleJai) [inline src/Search.tsx]: can we have it in class instead of inline

**Comment by invincibleJai** [comment]
@Meghanaskanagal can we have separate PR for pipelines

**Comment by rottencandy** [comment]
Closing as pipelines will need separate PR, and refactoring is required.

**Pull request closed without merging**